sql/src/main/java/org/opensearch/sql/sql/parser/AstExpressionBuilder.java

```diff
@@ -129,6 +130,17 @@ public UnresolvedExpression visitColumnName(ColumnNameContext ctx) {
     return visit(ctx.qualifiedName());
   }
 
+  @Override
+  public UnresolvedExpression visitArrayColumnName(ArrayColumnNameContext ctx) {
+    UnresolvedExpression qualifiedName = visit(ctx.qualifiedName());
+    if (ctx.decimalLiteral() == null) {
```

Comment:
Make `decimalLiteral` named value in ANTLR grammar, so you can reference the name here.